Make EditingRange use generated serialization

rniwa opened the pull request:
#### 63763f0db0b9e984701a87853719a8bc12339536
<pre>
Make EditingRange use generated serialization
<a href="https://bugs.webkit.org/show_bug.cgi?id=262641">https://bugs.webkit.org/show_bug.cgi?id=262641</a>

Reviewed by Alex Christensen.

Use the generated serialization for EditingRange.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/EditingRange.cpp:
(IPC::ArgumentCoder&lt;WebKit::EditingRange&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebKit::EditingRange&gt;::decode): Deleted.
* Source/WebKit/Shared/EditingRange.h:
* Source/WebKit/Shared/EditingRange.serialization.in: Added.

Canonical link: <a href="https://commits.webkit.org/268895@main">https://commits.webkit.org/268895@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb88bb3cc0c5e81d7e52e3ef425280b33eae551f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20983 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21367 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22049 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22864 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/19531 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24616 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21550 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21206 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20951 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/18206 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23719 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18112 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25310 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/19197 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19229 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23231 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19779 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19039 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/23351 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2594 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19614 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/DerivedSources.make:
```diff
@@ -501,6 +501,7 @@ SERIALIZATION_DESCRIPTION_FILES = \
 	Shared/CallbackID.serialization.in \
 	Shared/BackgroundFetchState.serialization.in \
 	Shared/DisplayListArgumentCoders.serialization.in \
+    Shared/EditingRange.serialization.in \
```
indentation

achristensen07 commented on Source/WebKit/Shared/EditingRange.serialization.in:
```diff
@@ -0,0 +1,31 @@
+}
+
+struct WebKit::EditingRange {
+    uint64_t location;
```
Maybe we should add a verifier that location + length doesn't overflow.

rniwa replied:
How does one add such a verifier?

achristensen07 replied:
Search for "Validator=" in the existing serialization.in files.

achristensen07 reviewed (approved):
Don't break the build

rniwa committed:
Make EditingRange use generated serialization
https://bugs.webkit.org/show_bug.cgi?id=262641

Reviewed by Alex Christensen.

Use the generated serialization for EditingRange.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/EditingRange.cpp:
(IPC::ArgumentCoder<WebKit::EditingRange>::encode): Deleted.
(IPC::ArgumentCoder<WebKit::EditingRange>::decode): Deleted.
* Source/WebKit/Shared/EditingRange.h:
* Source/WebKit/Shared/EditingRange.serialization.in: Added.

Canonical link: https://commits.webkit.org/268895@main
